controllers/kafkamanagement/kafkauser_controller.go

```diff
@@ -87,19 +87,40 @@ func (r *KafkaUserReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		Namespace: user.Spec.SecretRef.Namespace,
 	}
 	err = r.Client.Get(ctx, kafkaUserSecretNamespacedName, secret)
-	if k8serrors.IsNotFound(err) {
-		l.Error(err, "Cannot get Kafka user credentials. Secret is not found",
-			"request", req,
-		)
-		r.EventRecorder.Eventf(
-			user, models.Warning, models.FetchFailed,
-			"Fetch user credentials secret is failed. Reason: %v",
-			err,
-		)
+	if err != nil {
+		l.Error(err, "Cannot get secret for kafka user.", "request", req)
+		r.EventRecorder.Eventf(user, models.Warning, models.FetchFailed,
+			"Fetch user credentials secret is failed. Reason: %v", err)
 
 		return models.ReconcileRequeue, nil
 	}
 
+	if controllerutil.AddFinalizer(secret, user.GetDeletionFinalizer()) {
```

Comment:
I think it would be better to add finalizer after getting creds

Reply:
good idea! fixed